src/ApplePay/Decoding/PKCS7SignatureValidator.php

```diff
@@ -129,7 +129,22 @@ private function extractCertificates($signature)
 
         $certificates = $this->openSslService->getCertificatesFromPkcs7($pkcs7TemporaryFile->getPath());
 
-        return explode("\n\n", $certificates); // the response contains 2 certificates; separate them into an array
+        // The response contains 2 certificates. Let's extract them by their ends.
+        // It's good idea to ensure that certificates are at proper format. Some environments
+        // may ended up with additional "\n" characters.
+        //
+        $certificates = explode(
+            "-----END CERTIFICATE-----\n",
+            str_replace("\n\n", "\n", $certificates)
+        );
+
+        if (count($certificates) > 1) {
+            // Let's fix first certificate by adding extracted "end" signature.
+            //
+            $certificates[0] = $certificates[0] . '-----END CERTIFICATE-----';
```

Comment:
This fixes the first certificate but not the second one.
I think a more reliable fix would be to still use the initial explode but after that loop through the result and apply trim to all items. In case an item is "" (empty) after the trim, discard it. Would this work on your particular certificate?